services/movie.py

```
@@ -0,0 +1,36 @@
+from typing import Optional
+
+from django.db.models import QuerySet
+
+from db.models import Movie
+
+
+def get_movies(genres_ids: Optional[list[int]] = None,
+               actors_ids: Optional[list[int]] = None) -> QuerySet:
+    if (genres_ids is None) and (actors_ids is None):
+        return Movie.objects.all()
+    if genres_ids is not None and actors_ids is not None:
+        return Movie.objects.filter(
+            genres__id__in=genres_ids,
+            actors__id__in=actors_ids).distinct()
+    if genres_ids is not None and actors_ids is None:
+        return Movie.objects.filter(genres__id__in=genres_ids).distinct()
+    if genres_ids is None and actors_ids is not None:
+        return Movie.objects.filter(actors__id__in=actors_ids).distinct()
+
+
+def get_movie_by_id(movie_id: int) -> Optional[Movie]:
```

This type hint `Optional[Movie]` suggests that the function can return `None`. However, Django's `.get()` method raises a `DoesNotExist` exception if the object isn't found, rather than returning `None`. To more accurately reflect this behavior, the type hint should be just `Movie`.